charmcraft/config.py

```diff
@@ -71,6 +71,15 @@ def adapt_validation_error(error):
     raise CommandError(msg)
 
 
+@format_checker.checks('architecture', raises=ValueError)
+def check_architectures(value):
+    """Check the base architectures are valid charm-supported architectures."""
+    if isinstance(value, str):
+        if value in ["amd64", "arm64"]:
```

Comment:
Why other archs are not supported in the config? 

Charmcraft may not be able to build in other archs, but this is a decision to be made by the building command, the config does not know enough about the system to apply the restriction here.

Reply:
Just a starting point for the draft. Would you prefer an enumerated list (matching what's in ARCH_TRANSLATIONS), or just leave it unchecked?

Reply:
IMO there's no option, we must leave it unchecked. Think for example in this situation: two bases, one with `arch=[amd64]`, the other with `arch=[something-bizarre-that-will-work-in-a-private-builder]`, the charm config is totally fine to run on an `amd64` environment as "one base is valid".